File: ValuationCalculator.js
import React, { useState } from 'react';

function ValuationCalculator() {
  const [amountRaised, setAmountRaised] = useState('');
  const [equityPercentage, setEquityPercentage] = useState('');
  const [valuations, setValuations] = useState(null);

  const calculateValuations = () => {
    const amount = parseFloat(amountRaised);
    const equity = parseFloat(equityPercentage);
    const postMoneyValuation = amount / (equity / 100);
    const preMoneyValuation = postMoneyValuation - amount;
    setValuations({
      preMoney: preMoneyValuation.toFixed(2),
      postMoney: postMoneyValuation.toFixed(2)
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    calculateValuations();
  };

  return (
    <div>
      <h2>Startup Valuation Calculator</h2>
      <form onSubmit={handleSubmit}>
        <div>
          <label>
            Amount Being Raised ($):
            <input
              type="number"
              value={amountRaised}
              onChange={(e) => setAmountRaised(e.target.value)}
            />
          </label>
        </div>
        <div>
          <label>
            Equity Percentage (%):
            <input
              type="number"
              value={equityPercentage}
              onChange={(e) => setEquityPercentage(e.target.value)}
            />
          </label>
        </div>
        <button type="submit">Calculate</button>
      </form>
      {valuations && (
        <div>
          <p>Pre-Money Valuation: ${valuations.preMoney}</p>
          <p>Post-Money Valuation: ${valuations.postMoney}</p>
        </div>
      )}
    </div>
  );
}

export default ValuationCalculator;

File: valuationCalculator.py
def calculate_valuations(amount_raised, equity_percentage):
    """
    Calculate pre-money and post-money valuations.

    :param amount_raised: The amount of money the startup is raising.
    :param equity_percentage: The percentage of equity given away for the amount raised.
    :return: A tuple containing (pre_money_valuation, post_money_valuation)
    """
    post_money_valuation = amount_raised / (equity_percentage / 100)
    pre_money_valuation = post_money_valuation - amount_raised
    return pre_money_valuation, post_money_valuation

# Example usage
amount_raised = float(input("Enter the amount being raised (in dollars): "))
equity_percentage = float(input("Enter the equity percentage given away: "))

pre_money, post_money = calculate_valuations(amount_raised, equity_percentage)
print(f"Pre-Money Valuation: ${pre_money:,.2f}")
print(f"Post-Money Valuation: ${post_money:,.2f}")
